app/models/teams.js

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export class Teams {
+    static schema = {
+        name: 'Teams',
+        primaryKey: 'id',
+        properties: {
+            id: 'string',
+            createAt: 'int',
+            updateAt: 'int',
+            deleteAt: {type: 'int', default: 0},
+            displayName: 'string',
+            name: 'string',
+            type: 'string',
+            description: 'string?',
+            groupConstrained: {type: 'bool', default: false},
+            members: 'TeamMembers[]',
+        },
+    }
+}
+
+export class TeamMembers {
+    get team() {
+        if (this.teams?.length) {
+            return this.teams[0];
+        }
+
+        return null;
+    }
+
+    static schema = {
+        name: 'TeamMembers',
+        primaryKey: 'id', // the way we can update a nested object without having to specify the object to update we need a primary key
+        properties: {
+            id: 'string', // the primary key is {teamId}-{userId}
+            teams: {type: 'linkingObjects', objectType: 'Teams', property: 'members'},
```

Comment:
And here with the one-to-many link between TeamMembers and Teams

Reply:
same as above